velox/exec/tests/TestValueTest.cpp

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include "velox/exec/tests/utils/TestValue.h"
+
+namespace facebook::velox::exec::test {
+class TestObject {
+ public:
+  TestObject() = default;
+  ~TestObject() = default;
+
+  void set(int value) {
+    TestValue::adjust("facebook::velox::exec::test::TestObject::set", &value);
+    internalSet(value);
+  }
+
+ private:
+  void internalSet(int value) {
+    if (TestValue::enabled()) {
+      std::pair<int, int> testValue(value, count_);
+      TestValue::adjust(
+          "facebook::velox::exec::test::TestObject::internalSet", &testValue);
+    }
+    ++count_;
+    value_ = value;
+  }
+
+  int count_ = 0;
+  int value_;
+};
+
+TEST(TestValueTest, basic) {
+  for (const bool enabled : {false, true}) {
```

Comment:
I find this pattern difficult to read. Would it be clearer to split this test into two?